tool/tsh/config.go

```diff
@@ -19,44 +19,64 @@ package main
 import (
 	"fmt"
 	"net"
+	"os"
+	"os/exec"
+	"runtime"
 	"strings"
 	"text/template"
 
+	"github.com/gravitational/teleport/api/constants"
 	"github.com/gravitational/teleport/api/profile"
 	"github.com/gravitational/teleport/api/utils/keypaths"
 	"github.com/gravitational/trace"
 )
 
 const sshConfigTemplate = `
-# Common flags for all {{ .clusterName }} hosts
-Host *.{{ .clusterName }} {{ .proxyHost }}
-    UserKnownHostsFile "{{ .knownHostsPath }}"
-
-# Flags for all {{ .clusterName }} hosts except the proxy
-Host *.{{ .clusterName }} !{{ .proxyHost }}
+# Common flags for all {{ .ClusterName }} hosts
+Host *.{{ .ClusterName }} {{ .ProxyHost }}
+    UserKnownHostsFile "{{ .KnownHostsPath }}"
+    IdentityFile "{{ .IdentityFilePath }}"
+    CertificateFile "{{ .CertificateFilePath }}"
+
+# Flags for all {{ .ClusterName }} hosts except the proxy
+Host *.{{ .ClusterName }} !{{ .ProxyHost }}
     Port 3022
-    {{- if .leaf }}
-    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p@{{ .clusterName }}	
+    {{- if .Leaf }}
+    ProxyCommand {{ .TSHPath }} config-proxy --proxy={{ .ProxyHost }}:{{ .ProxyPort }} %h:%p "{{ .ClusterName }}"
     {{- else }}
-    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p
+    ProxyCommand {{ .TSHPath }} config-proxy --proxy={{ .ProxyHost }}:{{ .ProxyPort }} %h:%p
     {{- end }}
 `
 
+type hostConfigParameters struct {
+	ClusterName         string
+	KnownHostsPath      string
+	IdentityFilePath    string
+	CertificateFilePath string
+	ProxyHost           string
+	ProxyPort           string
+	Leaf                bool
+	TSHPath             string
+}
+
+// getSSHPath returns a sane default `ssh` path for the current platform.
+func getDefaultSSHPath() string {
+	if runtime.GOOS == constants.WindowsOS {
+		return "C:\\Windows\\System32\\OpenSSH\\ssh.exe"
+	}
+
+	return "ssh"
```

Comment:
This seems to be relying on PATH or shell evaluation somehow. That said I think POSIX mandates `/usr/bin/ssh` so that should be a sane default.

Reply:
That makes sense, fixed.